agent/agent.go

```diff
@@ -199,15 +219,36 @@ func (a *Agent) startCheckTicker() error {
 	}
 }
 
+func (a *Agent) startDiscoverTicker() error {
+	ticker := time.NewTicker(a.cfg.DiscoverInterval / 2)
+	// Repeat after 15 minutes
+	if a.cfg.DiscoverInterval < time.Minute {
+		a.cfg.DiscoverInterval = 15 * time.Minute
+	}
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ticker.C:
+			for i := 0; i < len(a.discovery); i++ {
+				a.discovery[i](a.consul)
+			}
+			a.consul.Agent().UpdateTTL("discover_cluster", "", consul.HealthPassing)
+		case <-a.ctx.Done():
+			close(a.wsResultChan)
+			return nil
+		}
+	}
+}
+
 func (a *Agent) updateConsulCheck(result CheckResult) {
 	log.Println("Updating Consul check TTL...")
 
 	var err error
 	var status string
 
-	summary := result.Summary()
+	summary := result.CheckSummary()
```

Comment:
I preferred the old `Summary` naming, but both are fine :)

Reply:
done